tool/scrape_mdn.dart

```diff
@@ -0,0 +1,239 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// ignore_for_file: prefer_expression_function_bodies
+
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:html/dom.dart' as dom;
+import 'package:html/dom_parsing.dart' show TreeVisitor;
+import 'package:html/parser.dart' show parse;
+import 'package:http/http.dart' as http;
+import 'package:pool/pool.dart';
+
+const apiUrl = 'https://developer.mozilla.org/en-US/docs/Web/API';
+
+Future<void> main(List<String> args) async {
+  final client = http.Client();
+
+  // Get the API page with all the interface references.
+  final response = await client.get(Uri.parse(apiUrl));
+  final doc = parse(response.body);
+
+  final section = doc.querySelector('section[aria-labelledby=interfaces]')!;
+  final anchorItems = section.querySelectorAll('li a');
+
+  final interfaceNames = <String>[];
+
+  for (final item in anchorItems) {
+    final href = item.attributes['href']!;
+    final interfaceName = href.split('/').last;
+
+    interfaceNames.add(interfaceName);
+  }
+
+  interfaceNames.sort();
+
+  print('${interfaceNames.length} items read from $apiUrl.');
+
+  final pool = Pool(6);
+
+  final interfaces = await pool.forEach(interfaceNames, (item) async {
+    return populateInterfaceInfo(item, client: client);
+  }).toList();
+
+  client.close();
+
+  const encoder = JsonEncoder.withIndent('  ');
+
+  final file = File('tool/mdn.json');
+  final json = {
```

Comment:
Is there any benefit to writing out the json and then consuming it later versus just having the node information in memory? If this is to just visualize the contents for now, then ignore this comment.

Reply:
Ah, I see, I didn't realize it was that slow. Then yes, let's avoid this slowing down iteration of the generation script if it's that significant.

I'm a little concerned on how to keep this updated (both in terms of the the script you added and the docs in the output files) if we aren't running the script to run the docs when we generate code regularly. This may not be a big issue after the original generation, but we're likely to have stale documentation. For this CL and the following ones, it's okay to not worry about this for now.